### PR TITLE
fix: commerce pages merge strategy not merging correctly

### PR DESCRIPTION
**Pull request by richard190m**
## Description

* Fix commerce pages merge strategy to be able to merge custom and base component, with the new structure with sections and subsections

## Checklist

- [X] The commit message follows our guidelines
- [X] Tests for the respective changes have been added
- [X] The code is commented, particularly in hard-to-understand areas
- [X] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by joaoprcosta — APPROVED** [review]

File: packages/redux/src/contents/utils.ts [hunk 1]
```diff
@@ -25,6 +25,40 @@ export enum ContentEnvironmentCode {
...
+ */
+const defaultContentReturn = {
+  number: 0,
+  totalItems: 0,
+  totalPages: 0,
+  entries: [],
+};
+
+/**
+ * Constant that represent the default section object returned by content api
```
Comment (boliveira) [inline packages/redux/src/contents/utils.ts:39]: This is a function so I would rephrase this to something like: "Function that returns the default section which would be returned by content api containing content generated by Content Tool"

File: packages/redux/src/contents/utils.ts [hunk 2]
```diff
@@ -162,7 +196,7 @@ export const getBestRankedCommercePageUsingDefaultStrategy = (
 
 /**
  * Method to return the best ranked commerce page with all components from other
- * pages without duplicates.
+ * pages without duplicates. This implementation only works with content from new Content Tool
```
Comment (boliveira) [inline packages/redux/src/contents/utils.ts:199]: Question: Will this be a breaking change for current consumers of this function?

Comment (richard190m) [reply packages/redux/src/contents/utils.ts:199]: This implementation is not used by any tenant. Currently, the only tenant that uses the merge strategy is Harrods, and they have their own implementation in their bff. This is a fix to BG implement merge strategy using blackout now

File: packages/redux/src/contents/utils.ts (hunk 1 above)
Comment (richard190m) [reply packages/redux/src/contents/utils.ts]: Fixed. Thanks

**Review by boliveira — APPROVED** [review]
